x-pack/plugins/synthetics/server/synthetics_service/project_monitor_formatter.ts

```diff
@@ -377,54 +379,101 @@ export class ProjectMonitorFormatter {
+              });
+            }
+            return !hasPrivateLocation;
           });
-        } else {
-          this.staleMonitors.push(staleMonitor.journeyId);
-          return null;
         }
       }
-    } catch (e) {
-      this.server.logger.error(e);
-    }
-  };
 
-  private deleteStaleMonitor = async ({
-    monitorId,
-    journeyId,
-  }: {
-    monitorId: string;
-    journeyId: string;
-  }) => {
-    try {
-      await deleteMonitor({
-        savedObjectsClient: this.savedObjectsClient,
-        server: this.server,
-        monitorId,
-        syntheticsMonitorClient: this.syntheticsMonitorClient,
-        request: this.request,
-      });
-      this.deletedMonitors.push(journeyId);
-      this.handleStreamingMessage({ message: `Monitor ${journeyId} deleted successfully` });
+      const chunkSize = 100;
+      for (let i = 0; i < monitors.length; i += chunkSize) {
+        const chunkMonitors = monitors.slice(i, i + chunkSize);
+        try {
+          if (!this.keepStale) {
+            await deleteMonitorBulk({
+              monitors: chunkMonitors,
+              savedObjectsClient: this.savedObjectsClient,
+              server: this.server,
+              syntheticsMonitorClient: this.syntheticsMonitorClient,
+              request: this.request,
+            });
+
+            for (const sm of chunkMonitors) {
+              const journeyId = (sm.attributes as MonitorFields)[ConfigKey.JOURNEY_ID]!;
+
+              this.deletedMonitors.push(journeyId);
+              this.handleStreamingMessage({
+                message: `Monitor ${journeyId} deleted successfully`,
+              });
+            }
+          } else {
+            chunkMonitors.forEach((sm) => {
+              const journeyId = (sm.attributes as MonitorFields)[ConfigKey.JOURNEY_ID]!;
+              this.staleMonitors.push(journeyId);
+            });
+          }
+        } catch (e) {
+          chunkMonitors.forEach((sm) => {
+            const journeyId = (sm.attributes as MonitorFields)[ConfigKey.JOURNEY_ID]!;
+
+            this.handleStreamingMessage({
+              message: `Monitor ${journeyId} could not be deleted`,
+            });
+            this.failedStaleMonitors.push({
+              id: journeyId,
+              reason: 'Failed to delete stale monitor',
+              details: e.message,
+              payload: staleMonitorsList.find(
+                (staleMonitor) => staleMonitor.savedObjectId === sm.id
+              ),
+            });
+          });
+          this.server.logger.error(e);
+        }
+      }
     } catch (e) {
-      this.handleStreamingMessage({ message: `Monitor ${journeyId} could not be deleted` });
-      this.failedStaleMonitors.push({
-        id: journeyId,
-        reason: 'Failed to delete stale monitor',
-        details: e.message,
-      });
+      this.server.logger.error(e);
```

Comment:
For bulk create and update, we sent back a message on the `this.failedMonitors` saying `x monitors failed to be created`. Why not add that here?

Reply:
basically there is an inner try/catch which does that for delete